onnxruntime/core/framework/utils.cc

```diff
@@ -633,12 +635,24 @@ void DumpNodeOutputs(OpKernelContext& context, const Node& node, const SessionSt
             if (!provider) {
               provider = cpu_execution_provider;
             }
+            std::cout << " Provider=" << provider->Type() << "\n";
 
+            const auto data_type = tensor.DataType();
             if (provider == cpu_execution_provider || tensor_location.mem_type == OrtMemTypeCPUOutput) {
-              const auto data_type = tensor.DataType();
               DispatchOnTensorType(data_type, DumpTensor, tensor, shape);
-            } else {
-              std::cout << " is not on CPU. Provider=" << provider->Type() << "\n";
+            } else if (provider == cuda_execution_provider) {
+              // copy tensor from gpu to cpu then dump
+              auto cpu_allocator = cpu_execution_provider->GetAllocator(0, OrtMemTypeDefault);
+              std::unique_ptr<Tensor> cpu_tensor = std::make_unique<Tensor>(data_type,
```

Comment:
>std::make_unique [](start = 51, length = 16)

we are trying to avoid std::make_unique, use onnxruntime::make_unique instead.

Reply:
it will break the CentOS build so you have to I believe.

---
In reply to: [338311925](https://github.com/microsoft/onnxruntime/pull/2243#discussion_r338311925) [](ancestors = 338311925)